streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1146,6 +1187,36 @@ private void verifyEOSTransactionTimeoutCompatibility() {
             configUpdates.put(COMMIT_INTERVAL_MS_CONFIG, EOS_DEFAULT_COMMIT_INTERVAL_MS);
         }
 
+        // validate rack awareness related configuration
+        final List<String> rackAwareAssignmentTags = getList(RACK_AWARE_ASSIGNMENT_TAGS_CONFIG);
```

Comment:
Not sure if this is the best place to perform those validations, but couldn't find any better place.

Reply:
Yes, put validation here is good.

Reply:
Maybe put the rack awareness validation into an another method. (If we decided to have these limitation)